lib/active_merchant/billing/gateways/redsys.rb

```diff
@@ -357,14 +361,17 @@ def add_stored_credential_options(data, options)
         case options[:stored_credential][:initial_transaction]
         when true
           data[:DS_MERCHANT_COF_INI] = 'S'
+          # recurring_cof_type = 'R'
         when false
           data[:DS_MERCHANT_COF_INI] = 'N'
           data[:DS_MERCHANT_COF_TXNID] = options[:stored_credential][:network_transaction_id] if options[:stored_credential][:network_transaction_id]
+          # recurring_cof_type = '0'
         end
 
         case options[:stored_credential][:reason_type]
         when 'recurring'
-          data[:DS_MERCHANT_COF_TYPE] = 'R'
+          data[:DS_MERCHANT_COF_TYPE] = "R"
+          # data[:DS_MERCHANT_COF_TYPE] = recurring_cof_type
```

Comment:
This conditionally set variable would satisfy the direction given by Banco Sabadell, but stored credentials tests start failing when this logic is included